src/metrics.go

```diff
@@ -10,122 +15,204 @@ import (
 	"github.com/newrelic/infra-integrations-sdk/log"
 )
 
-// getMetrics will gather all node and keyspace level metrics and return them as two maps
-// The main metrics map will contain all the keys got from JMX and the keyspace metrics map
-// Will contain maps for each <keyspace>.<columnFamily> found while inspecting JMX metrics.
-func getMetrics(client *gojmx.Client) (map[string]interface{}, map[string]map[string]interface{}, error) {
-	internalKeyspaces := map[string]struct{}{
+var (
+	// columnFamilyRegex matches the keyspace name and the scope.
+	columnFamilyRegex = regexp.MustCompile("keyspace=(.*),scope=(.*?),")
+
+	percentileRegex = regexp.MustCompile("attr=.*Percentile")
+
+	// filteredKeyspace set used to match internal keyspace that should not be reported.
+	filteredKeyspace = map[string]struct{}{
 		"OpsCenter":          {},
 		"system":             {},
 		"system_auth":        {},
 		"system_distributed": {},
 		"system_schema":      {},
 		"system_traces":      {},
 	}
+)
+
+// getMetrics will gather all node level metrics and return them as a map.
+func getMetrics(client *gojmx.Client, queryConfig []Query) (map[string]interface{}, error) {
 	metrics := make(map[string]interface{})
+
+	for _, query := range queryConfig {
+		attrNames := query.GetAttributeNames()
+
+		results, err := client.QueryMBeanAttributes(query.MBean, attrNames...)
+		if err != nil {
+			if jmxErr, ok := gojmx.IsJMXError(err); ok {
+				log.Debug("Failed to get attributes: %s, for mBeanName %s: %v", attrNames, query.MBean, jmxErr)
+				continue
+			}
+			return nil, fmt.Errorf("failed to perform query: %q: for attributes: %s error: %w", query.MBean, attrNames, err)
+		}
+
+		for _, jmxAttr := range results {
+			if jmxAttr.ResponseType == gojmx.ResponseTypeErr {
```

Comment:
is it normal that we can get a ResponseTypeErr without an error?

Reply:
yes, for e.g. we query for a batch of atttributes, the query is successfully, however only one value type is not supported.